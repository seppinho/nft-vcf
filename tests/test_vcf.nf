process TEST_VCF {

  input:
    path input_vcf

  output:
     path "output_*", emit: vcf


  """
	cp ${input_vcf} output_${input_vcf.name}
  """

}
