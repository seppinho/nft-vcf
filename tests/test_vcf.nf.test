nextflow_process {

    name "Test VCF"

    script "./test_vcf.nf"
    process "TEST_VCF"

    test("Should output samples") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.vcf.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            assert chromosome == "20"
            assert sampleCount == 51
            assert snpCount == 7824
            assert phased            
            //deprecated 
            assert noSamples == 51
            assert noSnps == 7824
        }
       }   
    }

    test("Should output number of SNPs") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.info.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            assert getNoSnps() == 63480
        }
       }   
    }

    test("Should output summary") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.info.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            assert snapshot(
              summary
            ).match()
        }
       }   
    }
    test("Should output header") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.info.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            assert snapshot(
              header.getColumnCount()
            ).match()
        }
       }   
    }

    test("Should output first 5 variants") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.info.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            assert snapshot(
              getVariants(5)
            ).match()
        }
       }   
    }
    test("Should output all variants") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.info.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            assert getVariants().size() == 63480
        }
       }   
    }

    test("Should output variants as MD5") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.info.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            assert snapshot(
              getVariantsMD5()
            ).match()
        }
       }   
    }
    test("Should output hetcount (path)") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.vcf.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            assert getVariant("20",68749).getHetCount() == 23 
        }
       }   
    }

    test("Should output hetcount (method)") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.vcf.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
            with(getVcfLine(path(filename),68749)) {
                assert getHetCount() == 23     
            }
       }   
    }

    test("Should output R2 info tag") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.info.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            assert getInfoR2("20",61795) == 0.567114
        }
       }   
    }

    test("Should output userdefined info tag") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.info.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            assert getInfoTag("R2","20",61795) == '0.567114'
        }
       }   
    }

    test("Should output VCF range") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.info.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            assert getRange("20",61795,64150).size() == 5
        }
       }   
    }    

    test("Should create index") {

        when {
          process {
          """
          input[0] = file("${projectDir}/tests/chr20.info.gz")
          """
          }
        }

        then {
          def filename = process.out.vcf.get(0)
          with(path(filename).vcf) {
            createIndex()
            assert workflow.success
        }
       }   
    } 
}
