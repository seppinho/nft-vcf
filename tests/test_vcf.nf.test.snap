{
    "Should output first 5 variants": {
        "content": [
            [
                "20\t61795\t.\tG\tT\t.\t.\tAF=0.252088;MAF=0.252088;AVG_CS=0.894225;R2=0.567114;IMPUTED",
                "20\t63231\t.\tT\tG\t.\t.\tAF=0.0370196;MAF=0.0370196;AVG_CS=0.984824;R2=0.703135;IMPUTED",
                "20\t63244\t.\tA\tC\t.\t.\tAF=0.150039;MAF=0.150039;AVG_CS=0.927431;R2=0.528218;IMPUTED",
                "20\t63799\t.\tC\tT\t.\t.\tAF=0.252088;MAF=0.252088;AVG_CS=0.894225;R2=0.567114;IMPUTED",
                "20\t64150\t.\tC\tA\t.\t.\tAF=0.0154804;MAF=0.0154804;AVG_CS=0.98452;R2=0.0776218;IMPUTED"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-11T14:47:49.276561"
    },
    "Should output variants as MD5": {
        "content": [
            "b6df00f466c4cd359a896aeab66e07b5"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-11T14:48:04.637136"
    },
    "Should output header": {
        "content": [
            8
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-30T12:46:36.528291"
    },
    "Should output summary": {
        "content": [
            "VcfFile [chromosomes=[20], noSamples=0, noSnps=63480, phased=true, phasedAutodetect=true, chrPrefix=false]"
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-30T12:45:23.226834"
    }
}